src/components/editor/AppController.jsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import { browserHistory } from 'react-router';
+import FalcorController from "lib/falcor/FalcorController";
+import { setAppReady } from "lib/falcor/falcorUtils";
+
+export default class AppController extends FalcorController {
+  constructor(props) {
+    super(props);
+    this.safeSetState({
+      defaultValue: this.handleDefaultValue.bind(this)()
```

Comment:
this defaultValue name is totally cryptic, I can't figure out what it does.

Also I believe this.function.bind(this) is redundant.